http-server/src/main/java/org/triplea/modules/game/ConnectivityController.java

```diff
@@ -38,11 +38,21 @@ public static ConnectivityController build() {
       keys = {KeyPart.IP},
       rates = {@Rate(limit = 10, duration = 1, timeUnit = TimeUnit.MINUTES)})
   @RolesAllowed(UserRole.HOST)
-  public boolean checkConnectivity(@Context final HttpServletRequest request, final Integer port) {
-    Preconditions.checkArgument(port > 0, "Port must be a positive number, was: " + port);
-    Preconditions.checkArgument(
-        port < Math.pow(2, 16), "Port must be less than max value (2^16), was: " + port);
+  public Response checkConnectivity(
+      @Auth final AuthenticatedUser authenticatedUser, final String gameId) {
 
-    return connectivityCheck.test(new InetSocketAddress(request.getRemoteAddr(), port));
+    final ConnectivityCheck.ReverseConnectionResult result =
+        connectivityCheck.canDoReverseConnect(authenticatedUser.getApiKey(), gameId);
+
+    switch (result) {
+      case SUCCESS:
+        return Response.ok().entity(true).build();
+      case FAILED:
+        return Response.ok().entity(false).build();
+      case GAME_ID_NOT_FOUND:
+        return Response.status(HttpStatus.UNPROCESSABLE_ENTITY_422).build();
+      default:
+        throw new IllegalStateException("Switch case not handled: " + result);
```

Comment:
Much better IMO 👍

Reply:
Agree, particularly getting the logic out of the controller.